src/app/Fake.DotNet.Cli/DotNet.fs

```diff
@@ -487,9 +487,17 @@ module DotNet =
         }
         static member Create() = {
             DotNetCliPath =
+                let version = try Some <| getSDKVersionFromGlobalJson() with _ -> None                              
                 findPossibleDotnetCliPaths None
-                |> Seq.tryHead
-                // shouldn't hit this one because the previous two probe PATH...
+                |> Seq.tryFind (fun cliPath -> 
+                    match version with 
+                    | Some version -> 
+                            version 
+                            |> Path.combine "sdk"
+                            |> Path.combine (Path.getDirectory cliPath)
+                            |> Directory.Exists
```

Comment:
Definitely an interesting approach. Do you think it is "OK" to have different behavior on CLI and fake running the command? Currently I think this should be opt-in but please feel free to tell me otherwise.

But honestly I think this could "just work".

Reply:
Having different behavior on CLI and fake in certainly doesn't feel great, but in this case that should only ever occur when there is a `global.json` containing a version that is not available on CLI, which always fails. 

I can't really imagine a case where it would cause trouble, but there are some cases where it would help the user not have to think about every aspect of how everything works, which I think is really one of the big strengths of Fake - some experienced people have made CI stuff much easier for less experienced people.

I do see the point of having "no magic", but sometimes it conflicts with having very simple APIs, and my personal point of view is that there is a middle ground.

I also see when trying to introduce Fake in my dayjob that these kind of details can make a big difference to getting new people excited about Fake.

Opt-in? Could be a good way to do it, any idea how that could look?

Reply:
Sorry I've been a little inactive here, but I've had the chance to think a little about this.

I totally understand your point of view, I'm just wondering if this is really adding any value - if the user already has to specify an option it would be just as easy to specify the install location provided by the install step, wouldn't it?
In that case this change is just adding complexity to the module, and in effect giving the users several methods of achieving the same thing (which is not great imo).

My suggestion is we just close this, then :)

Reply:
@severisv I think I'll leave that up to you, if you think this is valuable to get in, please fix the first comment and add some documentation around the behavior change and I'll merge this.
If not feel free to close this PR.

If we merge this an people are caught off-guard by this we can still change the behavior back ...

Reply:
I made the change you suggested (hopefully) :)

You also said to document it well, where and how do I document it exactly?

Reply:
I have no idea where this would fit, I guess the release notes are the way to go for now.